### PR TITLE
[CPDLP-3009] change ecf_id in NPQ from text to uuid

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

Ticket: https://dfedigital.atlassian.net.mcas.ms/browse/CPDLP-3009

### Changes proposed in this pull request

* Migration to changed `ecf_id` to `uuid` type.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -345,7 +345,7 @@
     t.string "email", default: "", null: false
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
-    t.text "ecf_id"
+    t.uuid "ecf_id"
```
Comment (cwrw) [inline db/schema.rb:348]: will this affect any indices? will it work ok with `nil` values? also can we test this on the migration env by deploying this just invase?

Comment (cwrw) [reply db/schema.rb:348]: also a few failing specs to consider :)

Comment (mooktakim) [reply db/schema.rb:348]: The indices should continue to work and it works with `nil` values. I've tested it locally.
Will run on migration env before prod merge.

File: db/schema.rb [hunk 2]
```diff
@@ -204,7 +204,7 @@
     t.text "name", null: false
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
-    t.text "ecf_id"
```
Comment (cwrw) [inline db/schema.rb:207]: can we change the factory to set up `SecureRandom.uuid`?

**Review by cwrw — COMMENTED** [review]
looks great, thanks Mook, just one last comment :)

File: db/schema.rb (hunk 2 above)
Comment (mooktakim) [reply db/schema.rb:207]: Updated now 👍

**Review by cwrw — APPROVED** [review]
Thanks @mooktakim! After testing on the migration env and deploying there happy for this to go in :)

**Pull request merged**